File: clients/ios/Lextures/Core/Routing/MobileProfileDepthPreferences.swift
import Foundation

/// Client rollout flag for M1.5 profile depth sections.
enum MobileProfileDepthPreferences {
    private static let key = "mobile_profile_depth"

    static var isEnabled: Bool {
        get {
            if UserDefaults.standard.object(forKey: key) == nil {
                return true
            }
            return UserDefaults.standard.bool(forKey: key)
        }
        set { UserDefaults.standard.set(newValue, forKey: key) }
    }
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/navigation/MobileProfileDepthPreferences.kt
package com.lextures.android.core.navigation

import android.content.Context

/** Client rollout flag for M1.5 profile depth sections. */
object MobileProfileDepthPreferences {
    private const val KEY = "mobile_profile_depth"

    fun isEnabled(context: Context): Boolean {
        val prefs = context.getSharedPreferences("lextures_mobile_ia", Context.MODE_PRIVATE)
        return if (!prefs.contains(KEY)) true else prefs.getBoolean(KEY, false)
    }

    fun setEnabled(context: Context, enabled: Boolean) {
        context.getSharedPreferences("lextures_mobile_ia", Context.MODE_PRIVATE)
            .edit()
            .putBoolean(KEY, enabled)
            .apply()
    }
}